pkg/service/model_build_target_group.go

```diff
@@ -276,7 +279,17 @@ func (t *defaultModelBuildTask) buildTargetGroupHealthCheckUnhealthyThresholdCou
 }
 
 func (t *defaultModelBuildTask) buildTargetType(_ context.Context) (elbv2model.TargetType, error) {
-	return elbv2model.TargetTypeIP, nil
+	var lbType string
+	_ = t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixLoadBalancerType, &lbType, t.service.Annotations)
+	var lbTargetType string
+	_ = t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixTargetType, &lbTargetType, t.service.Annotations)
+	if lbType == LoadBalancerTargetTypeNLBIP || (lbType == LoadBalancerTypeExternal && lbTargetType == LoadBalancerTargetTypeNLBIP) {
+		return elbv2model.TargetTypeIP, nil
+	}
+	if lbType == LoadBalancerTypeExternal && lbTargetType == LoadBalancerTargetTypeNLBInstance {
+		return elbv2model.TargetTypeInstance, nil
+	}
+	return "", errors.New("unsupported target type")
```

Comment:
nit: better to have lbType and lbTargetType in err message

Reply:
I'll change it.